### PR TITLE
feat(playground): add ability to load interactive examples from files

### PR DESCRIPTION
**Pull request by liamdebeasi**
This feature adds the following capabilities:

1. Demos can now be loaded from html files.

These samples are written using vanilla HTML + JavaScript and load Ionic from a CDN. As a best practice, we should fix the CDN versions to a particular major release. This is important because when we release Ionic 7, the playgrounds on the v7 docs should load Ionic 7, but the playgrounds on the v6 docs should load Ionic 6. This avoids the issue we have with the current phone demos where all versions of the phone demos load v6.

Each demo file should also load `common.js` and `common.css`. `common.js` allows for dynamic dark/light mode theming (see point 4 below). `common.css` adds the Ionic dark mode theme as well as utility classes for centering items in the playground.

As part of this I had to install a special html-loader so webpack was able to parse HTML files.

2. The playground can now be sized differently.

Open to other ideas, but I created pre-defined sizes (xsmall through xlarge) that can be used to set the size of the playground. I opted not to have the iframes get dynamically sized as a result of the inner content to avoid layout shifts. Thinking of a use case where we display a button that opens a large modal, I think it would be better to have the playground size stay fixed instead of resizing when the modal opens.

Developers can use one of the predefined sizes or pass in an explicit pixel value.

3. The playground can load iOS and MD samples.

This requirement was the main reason why I went with the iframe route. The mode configuration is not reactive in Ionic. This means that once you load the app in iOS mode, you cannot load it in MD mode without a full page reload. Having these demos be self contained in iframes allows us to show both iOS and MD mode samples.

4. The playground informs the demos of theme changes. 

Using [postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage), the Playground component now informs the iframes of light/dark mode settings so that the demo themes remain consistent with the rest of the Ionic Docs website.

### CODE REVIEW

**Review by amandaesmith3 — CHANGES_REQUESTED** [review]

File: static/usage/common.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * docs website theme.
+ */
+
+window.addEventListener('message', (ev) => {
+  const { data } = ev;
+
+  if (data.darkMode) {
+    document.body.classList.add('dark');
+  } else {
+    document.body.classList.remove('dark');
```
[the remark this answers is not in the document]
Comment (liamdebeasi) [reply static/usage/common.js:15]: What browser are you testing this on? I can't reproduce on my end.

File: src/components/global/Playground/index.tsx [hunk 2]
```diff
@@ -22,16 +25,53 @@ export default function Playground({
...
+   */
+  useEffect(async () => {
+    if (frameiOS.current && frameMD.current) {
+      if (!isIframeLoaded) {
+        await Promise.all([
+          waitForFrame(frameiOS.current),
+          waitForFrame(frameMD.current)
+        ]);
+
+        setIframeLoaded(true);
```
[the remark this answers is not in the document]
Comment (amandaesmith3) [reply src/components/global/Playground/index.tsx]: That works too 👍

File: static/usage/common.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (amandaesmith3) [reply static/usage/common.js:15]: Windows Chrome 99.0.4844.51. It's fairly consistent for me, for one or both calls: https://user-images.githubusercontent.com/90629384/158664658-6791d3ab-f959-40c5-a3bb-df2c9f36b43c.mp4

[the remark this answers is not in the document]
Comment (liamdebeasi) [reply static/usage/common.js]: Thanks! I can reproduce. I added a fix in https://github.com/ionic-team/ionic-docs/pull/2232/commits/c39b60ecf970b1444e279b6caab311281d1ca820

**Pull request merged**